compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1439,8 +1440,31 @@ public void visit(BLangErrorVariable varNode) {
             // We have no rhs to do type checking.
             return;
         }
+
         typeChecker.checkExpr(varNode.expr, env, varNode.getBType());
+        validateErrorDetailBindings(varNode);
+    }
+
+    private void validateErrorDetailBindings(BLangErrorVariable errorVariable) {
+        BType rhsType = types.getReferredType(errorVariable.expr.getBType());
+        if (rhsType.getKind() != TypeKind.ERROR) {
+            return;
+        }
+
+        BRecordType rhsDetailType = this.symbolEnter.getDetailAsARecordType((BErrorType) rhsType);
```

Comment:
Instead of calling this, if the detail type is a map, we can directly log errors for all the field binding patterns, right?